### PR TITLE
SDIT-1410 UserResource

### PR DESCRIPTION
**Pull request by steverendell**
Add role to /users/list endpoint and annotate the other /users endpoints with labels

### CODE REVIEW

File: src/main/java/uk/gov/justice/hmpps/prison/api/resource/ImageResource.java [hunk 1]
```diff
@@ -47,7 +47,7 @@ public class ImageResource {
             @ApiResponse(responseCode = "404", description = "Requested resource not found."),
             @ApiResponse(responseCode = "500", description = "Unrecoverable error occurred whilst processing request.")})
     @Operation(summary = "Image data (as bytes).", description = "Requires role VIEW_PRISONER_DATA.")
-    @PreAuthorize("hasRole('ROLE_VIEW_PRISONER_DATA')")
+    @PreAuthorize("hasRole('VIEW_PRISONER_DATA')")
```
Comment (andymarke) [inline src/main/java/uk/gov/justice/hmpps/prison/api/resource/ImageResource.java:50]: does this matter? do both forms work?

Comment (steverendell) [inline src/main/java/uk/gov/justice/hmpps/prison/api/resource/ImageResource.java:50]: Making it consistent - we normally omit ROLE_

File: src/main/java/uk/gov/justice/hmpps/prison/api/resource/UserResource.java [hunk 2]
```diff
@@ -113,43 +118,36 @@ public List<Location> getMyLocations(
...
-        try {
-            userService.setActiveCaseLoad(authenticationFacade.getCurrentUsername(), caseLoad.getCaseLoadId());
-        } catch (final AccessDeniedException ex) {
-            return ResponseEntity.status(HttpStatus.FORBIDDEN)
-                    .body(ErrorResponse.builder()
-                            .userMessage("Not Authorized")
-                            .developerMessage("The current user does not have acess to this CaseLoad")
-                            .build());
-        }
-        return ResponseEntity.ok().build();
```
Comment (steverendell) [inline src/main/java/uk/gov/justice/hmpps/prison/api/resource/UserResource.java:144]: redundant

File: src/main/java/uk/gov/justice/hmpps/prison/api/resource/UserResource.java [hunk 3]
```diff
@@ -158,17 +156,19 @@ public UserDetail getUserDetails(@PathVariable("username") @Parameter(descriptio
 
     @ApiResponses({@ApiResponse(responseCode = "200", description = "The list of user details")})
     @Operation(summary = "Returns the user details for supplied usernames - POST version to allow large user lists.", description = "user details for supplied usernames")
+    @PreAuthorize("hasRole('STAFF_SEARCH')")
```
Comment (steverendell) [inline src/main/java/uk/gov/justice/hmpps/prison/api/resource/UserResource.java:159]: The only actual change of behaviour

File: src/main/java/uk/gov/justice/hmpps/prison/api/resource/UserResource.java [hunk 4]
```diff
@@ -158,17 +156,19 @@ public UserDetail getUserDetails(@PathVariable("username") @Parameter(descriptio
...
     @ApiResponses({
-            @ApiResponse(responseCode = "200", description = "No New Users", content = {@Content(mediaType = "application/json", schema = @Schema(implementation = CaseloadUpdate.class))}),
-            @ApiResponse(responseCode = "201", description = "New Users Enabled", content = {@Content(mediaType = "application/json", schema = @Schema(implementation = CaseloadUpdate.class))}),
+        @ApiResponse(responseCode = "200", description = "No New Users", content = {@Content(mediaType = "application/json", schema = @Schema(implementation = CaseloadUpdate.class))}),
+        @ApiResponse(responseCode = "201", description = "New Users Enabled", content = {@Content(mediaType = "application/json", schema = @Schema(implementation = CaseloadUpdate.class))}),
     })
     @Operation(summary = "Add the NWEB caseload to specified caseload.", description = "Add the NWEB caseload to specified caseload.")
-    @PutMapping("/add/default/{caseload}")
+    @PutMapping("/add/default/{caseload}") // This is the only endpoint which is only called with client_credentials
+    @PreAuthorize("hasAnyRole('MAINTAIN_ACCESS_ROLES','MAINTAIN_ACCESS_ROLES_ADMIN')")
```
Comment (steverendell) [inline src/main/java/uk/gov/justice/hmpps/prison/api/resource/UserResource.java:171]: moved from service

File: src/main/java/uk/gov/justice/hmpps/prison/repository/sql/CaseNoteRepositorySql.kt [hunk 5]
```diff
@@ -4,15 +4,15 @@ enum class CaseNoteRepositorySql(val sql: String) {
...
         MAX(OCS.CONTACT_TIME) LATEST_CASE_NOTE
-                FROM OFFENDER_CASE_NOTES OCS JOIN OFFENDER_BOOKINGS OB
-                ON OB.OFFENDER_BOOK_ID = OCS.OFFENDER_BOOK_ID JOIN OFFENDERS O
-        ON O.OFFENDER_ID = OB.OFFENDER_ID
-                WHERE OCS.AUDIT_TIMESTAMP between :fromDate and :toDate
+      FROM OFFENDER_CASE_NOTES OCS
+        JOIN OFFENDER_BOOKINGS OB ON OB.OFFENDER_BOOK_ID = OCS.OFFENDER_BOOK_ID
+        JOIN OFFENDERS O ON O.OFFENDER_ID = OB.OFFENDER_ID
+      WHERE OCS.AUDIT_TIMESTAMP between :fromDate and :toDate
         %s
```
Comment (steverendell) [inline src/main/java/uk/gov/justice/hmpps/prison/repository/sql/CaseNoteRepositorySql.kt:16]: just corrected formatting

File: src/main/java/uk/gov/justice/hmpps/prison/service/UserService.java [hunk 6]
```diff
@@ -116,15 +115,10 @@ public List<UserRole> getRolesByUsername(final String username, final boolean al
             .toList();
     }
 
-    public boolean isUserAccessibleCaseloadAvailable(final String caseload, final String username) {
-        return userCaseloadRepository.findById(UserCaseloadId.builder().caseload(caseload).username(username).build()).isPresent();
-    }
```
Comment (steverendell) [inline src/main/java/uk/gov/justice/hmpps/prison/service/UserService.java:121]: not used

File: src/main/java/uk/gov/justice/hmpps/prison/api/resource/ImageResource.java (hunk 1 above)
Comment (andymarke) [reply src/main/java/uk/gov/justice/hmpps/prison/api/resource/ImageResource.java:50]: cool

File: src/test/java/uk/gov/justice/hmpps/prison/executablespecification/AbstractStepDefinitions.java [hunk 7]
```diff
@@ -98,11 +97,6 @@ public ReferenceDomainsSteps referenceDomain() {
             return new ReferenceDomainsSteps();
         }
 
-        @Bean
-        public MyAssignmentsSteps userAssignment() {
-            return new MyAssignmentsSteps();
-        }
```
Comment (steverendell) [inline src/test/java/uk/gov/justice/hmpps/prison/executablespecification/AbstractStepDefinitions.java:104]: not used

**Review by andymarke — APPROVED** [review]
LGTM

File: src/main/java/uk/gov/justice/hmpps/prison/api/resource/ImageResource.java (hunk 1 above)
Comment (steverendell) [reply src/main/java/uk/gov/justice/hmpps/prison/api/resource/ImageResource.java:50]: Maybe Im being OCD !

**Pull request merged**